test/jit/test_dtype_analysis.py

```diff
@@ -19,8 +31,59 @@
     )
 
 
-# XXX: Relies on Symbolic Shape Analysis, which is still a prototype
-class TestDtypeAnalysis(JitTestCase):
+custom_rules_works_list = {
```

Comment:
`custom_rules_works_list` and `custom_rules_works_list` should cover all custom rules we borrowed from the old shape analysis

Reply:
We are still relying on shape analysis for shape analysis (for knowing when we have zero-dim tensors)